<launch>
    <arg name="sim_arm" default="true" />
    <arg name="sim_sensor" default="true" />
    <arg name="arm_id" default="arm1" />
    <arg name="tool_name" default="bracket" />

    <!-- Arm -->
    <group if="$(arg sim_arm)">
         <!-- MoveIt only (fake execution) -->
        <include file="$(find mag_device_arm)/launch/fr5/single/fr5_single_moveit.launch">
            <arg name="arm_id" value="$(arg arm_id)" />
            <arg name="tool_name" value="$(arg tool_name)" />
        </include>
    </group>

    <group unless="$(arg sim_arm)">
         <include file="$(find mag_device_arm)/launch/fr5/single/fr5_single_bringup.launch">
            <arg name="arm_id" value="$(arg arm_id)" />
            <arg name="tool_name" value="$(arg tool_name)" />
         </include>
    </group>

    <!-- Sensor -->
    <group if="$(arg sim_sensor)">
        <include file="$(find mag_device_sensor)/launch/simulation.launch" />
    </group>
    <group unless="$(arg sim_sensor)">
        <include file="$(find mag_device_sensor)/launch/hardware.launch" />
    </group>

    <!-- Scanner Node -->
    <node pkg="mag_planar_scan" type="planar_scan_node" name="planar_scan_node" output="screen">
        <param name="arm_name" value="$(arg arm_id)" />
        <rosparam command="load" file="$(find mag_planar_scan)/config/planar_scan_config.yaml" />
    </node>
</launch>
